### PR TITLE
feat: add find answers query, endpoint, controller and update ticket resolver

### PR DESCRIPTION
**Pull request by wardner**
#### What does it do?

Update the yarn lock file

Adds the find answers query to the custom-fields micro-service, with the endpoint in the rpc server, the controller and validator.  With the unit tests needed for the query.

Update the helpdesk-api to add the answers data source, create the answer model and update the ticket resolver to add answers to get ticket details.

#### Why is it needed?

To show the answers value in the helpdesk-api, when the agent receives a ticket detail.

#### How to test it?

`Scenario 1:` Run unit tests

1. Go to custom-fields unit tests folder
2. In the readme, copy the line that runs the unit tests
3. Paste it in the console and run it

**Result**: It should run the tests and pass it without problems

`I agree`

Before asking for review I...

- [x] Used this first label: `🕐 Peer Review Pending`
- [x] Have added all relevant information about my changes in this PR
- [x] Have performed a self-review of my own code
- [x] Have made sure my code follows the project's style guidelines
- [x] Have checked my code and corrected any misspellings
- [x] Have made sure my changes generate no new warnings

/cc @efraa

### CODE REVIEW

File: apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQuery.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Licensed under BSD 3-Clause License. See LICENSE for terms.
+ */
+
+import type { Extra, Guid } from '@turnly/common'
+import type { IQuery } from '@turnly/shared'
+
+export class FindAnswersQuery implements IQuery {
+  public constructor(
+    public readonly organizationId: Guid,
+    public readonly entityType: string,
```
Comment (efraa) [inline apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQuery.ts:14]: Please add an EntityType enum.

```typescript
enum EntityType {
  CUSTOMER = 'customer',
  TICKET = 'ticket'
}
```

File: apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQueryHandler.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    entityType,
+    fieldId,
+    extra,
+  }: FindAnswersQuery) {
+    const query = new QueryBuilder<Answer>()
+      .equal('organizationId', organizationId)
+      .equal('entityType', entityType)
+
+    if (fieldId) query.equal('fieldId', fieldId)
+    if (extra) {
```
Comment (efraa) [inline apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQueryHandler.ts:33]: Use the `?` operator for this condition and remove the `if`

```typescript
      extra?.forEach(e => {
        query.inExtra(e.key, e.value)
      })
```

**Review by efraa — APPROVED** [review]

File: apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQuery.ts (hunk 1 above)
Comment (wardner) [reply apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQuery.ts]: Done

File: apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQueryHandler.ts (hunk 2 above)
Comment (wardner) [reply apps/custom-fields/src/Answers/application/queries/FindAnswersQuery/FindAnswersQueryHandler.ts]: Done

**Pull request merged**